### ISSUE TITLE
Root detection not working on android 14 

### ISSUE THREAD

**Issue by pravinlondhe** [scottyab/rootbeer#221]
Root detection not working on android 14 emulator google API's system image while on android 13 emulator google API's system image it is working.

**Comment by NityaSantosh26** [comment]
@pravinlondhe, Yes I also have this issue. The root detection doesn't work from Android 14 emulator onwards (Also tested on Android 15 Emulator). 
I have digged deeper into the code and the failure is seems to be with https://github.com/scottyab/rootbeer/blob/master/rootbeerlib/src/main/java/com/scottyab/rootbeer/RootBeer.java, checkForDangerousProps() method.
The properties ro.debuggable, ro.secure are not part of the properties list of the emulator (Android 14)
Leaving that aside, I expected checkForBinary(BINARY_SU) method to return true but this fails in verifying if the su file exists.
The file indeed exists in the device explorer but the Line 191 of Rootbeer.java fails due to permission is not granted.
I am not sure about the real rooted device scenario, but on the Android 14 & 15 rooted emulators are not been detected!
@scottyab, Please look into it. Atleast the checkForBinary going fine. Even tried checkSuExists() which also fails detecting rooted device.

**Comment by scottyab** [comment]
Thanks both for flagging this. Just so I'm clear what your are reporting. The only reason Android 13 emulator was flagging as rooted was the `checkForDangerousProps()` check specifically  `ro.debuggable`, `ro.secure` and new on android 14/15 emulator the properties are not set/accessible?  Or were there other root checks that returned false on Android 13? 

> The file indeed exists in the device explorer

@NityaSantosh26 What path in the su binary that you're seeing is not detected? I'm unclear if this is a file you've added for test purposes or something you've manually added to the emulator.

**Comment by NityaSantosh26** [comment]
Thanks for your reply @scottyab.
That's correct only with the check for _ro.debuggable, ro.secure_ the root detection is being done on Android 13 Emulator. All the other properties/methods return false.
For Android 14 and above, the two properties are not available in the list of all properties maybe they are removed (i am not sure).
So even this checkForDangerousProps() also fails resulting in no root detection.

**Comment by scottyab** [comment]
I'm seeing `ro.force.debuggable` rather than `ro.debuggable` in system properties from Android 14 and Android 15, it could be this changed in Android 14 although not found any proof of that yet. Even if we changed it to also check for `ro.force.debuggable`, this is `[0]` so it would still pass the check and not be detected as rooted. 

When I test the Default Android System image (i.e not the recommended Google one) emulator I'm setting test keys flagged and so rooted check fails. 

![image](https://github.com/user-attachments/assets/94bbf495-cd9f-4166-96b4-27a47a0f68f6)


I'm not sure of the best solution here, potentially it's a clarification about the expected root check result of the emulator for the default system image vs google system images?   
 


We could add a check for this property and if  is as discussed in another issue, ro.debuggable isn't really an indicator of root.

**Comment by NityaSantosh26** [comment]
Yeah. But how about checkForBinary?

**Comment by yeshwanthsai1010** [comment]
Hi @scottyab 
I looked into this issue and confirmed that starting with Android 14, ro.debuggable and ro.secure properties are no longer accessible ([reference](https://issuetracker.google.com/issues/298287672)). This has caused root detection to fail in emulators.

To address this, I found that the ro.adb.secure property can be used as an alternative for detecting rooted devices(Emulators). This approach is supported by a discussion on [XDA forums](https://xdaforums.com/t/ro-debuggable-1-but-rooted-debugging-does-not-show-up.4686059/). I’ve updated the root detection logic accordingly and submitted a [Pull Request](https://github.com/scottyab/rootbeer/pull/242)

Please review and approve the pull request so we can restore root detection functionality for Android 14+ emulators. Let me know if you need any further details!
